### ISSUE TITLE
What's changed?

### ISSUE THREAD

**Issue by 0seba** [OpenBMB/VoxCPM#109]
Hi, thank you and congratz on the updates to the model and the codebase.

I used the updated codebase to generate with the old model VoxCPM 0.5B and I was surprised that it's stability for long term generation improved a lot, generating several minutes of clearly understandable speech, unlike the previous version which began to have issues over 20 seconds of audio.

I tried to look for the changes that allow for this to be possible so I can apply those changes to the Apple Neural Engine inference code I maintain, but I was not able to pinpoint which fixes where introduced in the code that allowed for this improvement. This is the commit I am comparing against to look for the changes d1bb6aaf41f5f1b9febba4d2f32ac0850bf500d0.

Could you be able to provide me some pointers to which where these changes that allow to improve the generation quality for both the old and new models?

Thanks again!

Edit: played around a bit more and seems like it was the TextNormalizer and disabling fixed the problem

**Comment by Labmem-Zhouyx** [comment]
Hi,

Thank you for your feedback! The improved stability for long-generation likely comes from a key change in the audio preprocessing during inference.

We switched from right-padding to left-padding when the prompt audio length isn’t divisible by patch_len (at https://github.com/OpenBMB/VoxCPM/blob/main/src/voxcpm/model/voxcpm.py#L406):

```
if audio.size(1) % patch_len != 0:
    padding_size = patch_len - audio.size(1) % patch_len
    audio = torch.nn.functional.pad(audio, (padding_size, 0))  # left padding
```
This ensures valid prompt audio stays at the end of the sequence, which is crucial for maintaining coherence in autoregressive generation.

Additionally, we have removed the previous trim trick of the last patch in prompt_audio. You may further examine the generate() and inference() functions in src/voxcpm/model/voxcpm.py for more details. :)

Thanks again for testing and your contributions!
